builtin/providers/aws/resource_aws_instance.go

```diff
@@ -986,7 +997,15 @@ func buildAwsInstanceOpts(
 		}
 	}
 
-	if hasSubnet && associatePublicIPAddress {
+	if hasNetworkInterface {
+		ni := &ec2.InstanceNetworkInterfaceSpecification{
+			NetworkInterfaceId: aws.String(networkInterfaceID),
+			DeviceIndex:        aws.Int64(int64(0)),
+			SubnetId:           aws.String(subnetID),
```

Comment:
On that note, if we go that path, what's the best way to validate that certain attributes aren't specified together? `ConflictsWith` won't work because they are `Computed` attributes as well.